providers/ebs_raid.rb

```diff
@@ -296,15 +297,25 @@ def attach_volume(disk_dev, volume_id)
   Chef::Log.info("Attaching existing ebs volume id #{volume_id} for device #{disk_dev_path}")
 
   creds = aws_creds # cannot be invoked inside the block
-  aws_ebs_volume disk_dev_path do
-    aws_access_key creds['aws_access_key_id']
-    aws_secret_access_key creds['aws_secret_access_key']
-    aws_session_token creds['aws_session_token']
-    device disk_dev_path
-    name disk_dev
-    volume_id volume_id
-    action [:attach]
-    provider 'aws_ebs_volume'
+  if creds.nil? then
+     aws_ebs_volume disk_dev_path do
+       device disk_dev_path
+       name disk_dev
+       volume_id volume_id
+       action [:attach]
+       provider 'aws_ebs_volume'
+     end
+  else
+     aws_ebs_volume disk_dev_path do
+       aws_access_key creds['aws_access_key_id']
```

Comment:
@coppick could you avoid the duplication and the if block by putting a conditional on these attributes?

```
aws_ebs_volume disk_dev_path do
  aws_access_key creds['aws_access_key_id'] unless creds.nil?
  aws_secret_access_key creds['aws_secret_access_key'] unless creds.nil?
  aws_session_token creds['aws_session_token'] unless creds.nil?
  device disk_dev_path
  name disk_dev
  volume_id volume_id
  action [:attach]
  provider 'aws_ebs_volume'
end
```

OR

```
aws_ebs_volume disk_dev_path do
  unless creds.nil? do
    aws_access_key creds['aws_access_key_id']
    aws_secret_access_key creds['aws_secret_access_key']
    aws_session_token creds['aws_session_token']
  end
  device disk_dev_path
  name disk_dev
  volume_id volume_id
  action [:attach]
  provider 'aws_ebs_volume'
end
```

Of course my Chef is a little rusty and I didn't test this, but I think this is doable.